feat(DEV-15064): PayablesTable use Payment Records

dinis-monite opened the pull request:
# Scope

Change Payables Table to show payment actions based on Payment Intents.

Jira ticket: https://monite.atlassian.net/browse/DEV-15064

# Implementation

- Updated API schema file, so that payment_records endpoint has `object_ids_in` filter.
- Added new hook `usePayablePaymentIntentsAndRecords` to fetch and process payment records for payables.
- Refactored `PayablesTable` with columns Total, Paid, and Due.
- Changed `PayablesTableActions` to use payment intents to show action on `PayablesTable`.
- Improved `usePaymentHandler` to create link or fetch existing from passed intent ID.
   - previous behavior would always create a new payment link every time the pay button was pressed.
-  Removed `isPaymentLinkAvailable` from PayableDetails components
   - `isPaymentLinkAvailable` was removed from `usePaymentHandler`, since the value was hardcoded as true.
- Changed `usePaymentIntent` to enable query only when payment intent ID is provided.


# Remarks

The update to the `schema.json` file to get new definitions and schemas related to `payment_records` was *made manually*, because the `yarn download-openapi-document` command script pulled lots of changes, that break typechecks.
~Currently, the typechecks are failing with this changes, since the update to API schema file pulled lots of changes. This PR is ready for review, but that problem needs to be solved before merging.~

Partial payment are out of the scope of this task.


# Steps to test

On the table of the Payables screen:
- there should be columns Total, Paid, and Due, showing the Payable data.
- When the Payable is in status "waiting_to_be_paid" or "partially_paid", the "payment column" shows payment-related actions or status:
  - No "pay" button if user has no permissions to pay.
  - No "pay" button if payable has no amount to pay.
  - "Pay" button if no payments records exist .
     - clicking the "pay" button should create a new payment link (check devtools for POST request) and open the modal with the Payment page.
  - "Pay" button if ONLY payment records in status "created".
     - clicking the "pay" button should NOT create a new payment link (check devtools for POST request), should load the existing payment intent/link, and open the modal with the Payment page.
  - If a scheduled payment records exists, should show a chip with text "scheduled" and the date.
     - Hovering should show a tooltip with more details (i.e. amount).
  - If x processing payment records exist, should show a chip with text "x processing".
     - Hovering should show a tooltip with more details (i.e. sum of the amounts).
  - If a success payment records exists, should show a chip with text "settled".
- Partial payment are out of the scope of this task.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
  - Enhanced the Payables Table with three new columns: Total, Due, and Paid, providing clearer visibility into payment amounts.
  - Payment actions in the Payables Table are now powered by Payment Intents, offering improved status tracking and user feedback.
  - Added detailed UI indicators for scheduled, processing, and succeeded payments within the Payables Table.
  - Introduced a new "Pay" button visibility logic based on payment intents and user permissions.
- **Bug Fixes**
  - Improved error handling and messaging for payment link failures.
  - Fixed date display in payment records to prevent showing invalid dates.
- **Documentation**
  - Updated localization with new messages for payment statuses and actions.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

caiomachado-monite commented on packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx:
```diff
@@ -341,6 +356,49 @@ const PayablesTableBase = ({
+            );
+          }
+
+          return <span style={{ opacity: 0.4 }}>-</span>;
```
Same as the comment above

caiomachado-monite commented on packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx:
```diff
@@ -353,27 +411,31 @@ const PayablesTableBase = ({
-              params.value
+              formatCurrencyToDisplay(params.value, params.row.currency)
             ) : (
               <span style={{ opacity: 0.4 }}>0.00</span>
```
I know you didn't change it but I think we can update it here as well with the tailwind class

dinis-monite commented on packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx:
```diff
@@ -214,6 +215,22 @@
       : undefined,
   });
 
+  // Get IDs of payables in status [waiting_to_be_paid, partially_paid]
+  const payableIdsInWaitingToBePaidOrPartiallyPaid = useMemo(() => {
+    return payables?.data
+      ?.filter(
+        (payable) =>
+          payable.status === 'waiting_to_be_paid' ||
+          payable.status === 'partially_paid'
+      )
+      .map((payable) => payable.id);
+  }, [payables?.data]);
+  // Fetch payment records for payables in status [waiting_to_be_paid, partially_paid]
+  const { payablesPaymentIntentsRecord, isLoading: isPaymentRecordsLoading } =
+    usePayablePaymentIntentsAndRecords(
+      payableIdsInWaitingToBePaidOrPartiallyPaid || []
+    );
+
```
The check is already implemented in `usePayablePaymentIntentsAndRecords`, which avoids the request if the passed array is empty

dinis-monite commented:
Updated the PR to unblock the typecheck errors.
Reverted my previous commit that pulled all the API schema changes and made a [new commit where I **manually copy** the changes related to payment records](https://github.com/team-monite/monite-sdk/pull/738/commits/53bb29bee58c9d25a0b09c25012affc1251d6af2)
